tests/unit-core/test-api.c

```diff
@@ -1169,6 +1169,20 @@ main (void)
                                                                       global_mode_snapshot_buffer,
                                                                       SNAPSHOT_BUFFER_SIZE);
     TEST_ASSERT (global_mode_snapshot_size != 0);
+
+    /* Check the snapshot data. Unused bytes should be filled with zeroes */
+    const uint8_t expected_data[] =
+    {
+      0x07, 0x00, 0x00, 0x00, 0x38, 0x00, 0x00, 0x00, 0x20, 0x00, 0x00, 0x00, 0x01,
+      0x00, 0x00, 0x00, 0x03, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x00, 0x00,
+      0x00, 0x01, 0x05, 0x00, 0x28, 0x00, 0xB7, 0x46, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x02, 0x00, 0x01, 0x00, 0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x01,
+      0x01, 0x00, 0x47, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x14,
+      0x00, 0x73, 0x74, 0x72, 0x69, 0x6E, 0x67, 0x20, 0x66, 0x72, 0x6F, 0x6D, 0x20,
+      0x73, 0x6E, 0x61, 0x70, 0x73, 0x68, 0x6F, 0x74, 0x00, 0x00
```

Comment:
What about big endian systems?

Reply:
Hmm, I don't know... Any suggestions?
Could use `#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__` etc.
If possible I'd rather only test for the zero padding, but I don't see a way to do that w/o using the internal structures here directly.